### PR TITLE
Removing most post install processing of Android plugin

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -140,23 +140,31 @@
...
-          <category android:name="android.intent.category.LAUNCHER" />
-        </intent-filter>
-      </activity>
-    </config-file>
+          android:theme="@android:style/Theme.Black.NoTitleBar"
+          android:launchMode="singleTop" />
+    </edit-config>
+
+    <!-- Fixes the application tag in the Manifest -->
+    <edit-config file="AndroidManifest.xml" target="/manifest/application" mode="merge">
```
Comment (bhariharan) [inline plugin.xml]: `edit-config` for inline `Application` replacement.

File: plugin.xml [hunk 2]
```diff
@@ -140,23 +140,31 @@
...
+      <application android:hardwareAccelerated="true"
+          android:icon="@drawable/sf__icon"
+          android:label="@string/app_name"
+          android:manageSpaceActivity="com.salesforce.androidsdk.ui.ManageSpaceActivity"
+          android:name="com.salesforce.androidsdk.phonegap.app.HybridApp" />
+    </edit-config>
 
-    <config-file target="res/xml/config.xml" parent="/*">
+    <!-- Fixes the min and target SDK versions in the Manifest -->
+    <edit-config file="AndroidManifest.xml" target="/manifest/uses-sdk" mode="merge">
```
Comment (bhariharan) [inline plugin.xml]: `edit-config` for right `minSdkVersion` and `targetSdkVersion`. This seems more reliable than using `<preference>` tags.

**Comment by bhariharan** [comment]
Generated hybrid app works fine out of the box with `Android Studio` (just like before). I haven't attempted complicated use cases like removing and re-adding our plugin, etc.

**Comment by bhariharan** [comment]
Command line compilation (`./gradlew assembleDebug`) of the generated app works fine too.

**Comment by bhariharan** [comment]
@wmathurin Great news - you can now remove and re-add the plugin and run `cordova prepare` as many times as you want. Appears to work fine.

File: plugin.xml [hunk 3]
```diff
@@ -181,9 +189,6 @@
     </config-file>
 
     <hook type="after_plugin_install" src="tools/postinstall-android.js" />
-    <hook type="after_platform_add" src="tools/after-platform-add-android.js" />
```
Comment (wmathurin) [inline plugin.xml]: Awesome!

**Pull request merged**